core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -572,6 +572,12 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
+  private[spark] val DYN_ALLOCATION_REUSE_EXECUTORS =
+    ConfigBuilder("spark.dynamicAllocation.reuseExecutors")
```

Comment:
I think this is to generic and could confuse users, I would prefer to see something in the name specifically using resource profiles.  Really I think I would prefer it to be scheduler config like:
spark.scheduler.reuseCompatibleExecutors
 version below needs updating